packages/theme/src/cli/utilities/theme-environment/storefront-session.ts

```diff
@@ -2,16 +2,48 @@ import {parseCookies, serializeCookies} from './cookies.js'
 import {defaultHeaders} from './storefront-utils.js'
 import {fetch} from '@shopify/cli-kit/node/http'
 import {AbortError} from '@shopify/cli-kit/node/error'
+import {createInterface} from 'readline'
 
 export async function isStorefrontPasswordProtected(storeURL: string): Promise<boolean> {
-  const response = await fetch(storeURL, {
+  const response = await fetch(ensureHttps(storeURL), {
     method: 'GET',
     redirect: 'manual',
   })
 
   return response.status === 302
 }
 
+export function promptPassword(prompt: string): Promise<string> {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  })
+
+  return new Promise((resolve) => {
+    readline.question(prompt, (password) => {
+      readline.close()
+      resolve(password)
+    })
+  })
+}
```

Comment:
Why are we prompting with `readline` instead of using one of the prompts defined in `cli-kit`?
We should be consistent with the style CLI prompts

Reply:
Tinkering with using `renderTextPrompt` for the REPL loop. 2 questions:

<img width="700" alt="image" src="https://github.com/user-attachments/assets/d94aa4f6-99a2-4e7e-8757-8a83dbe147b6">

1) Is there a way to remove the `:`? [Seems like it needs to end with punctation](https://github.com/Shopify/cli/blob/7b53b6dd8ff049f233b773b0a609cd25522db3ef/packages/cli-kit/src/private/node/ui/utilities.ts#L5) but I would like to render something like the screenshot below
2) _This one is less important_ - Is there a way to modify `?` to use `>` instead? [source](https://github.com/Shopify/cli/blob/7b53b6dd8ff049f233b773b0a609cd25522db3ef/packages/cli-kit/src/private/node/ui/components/TextPrompt.tsx#L94)

Goal
![image](https://github.com/user-attachments/assets/b80f08da-eb54-43bc-ad2f-335706f6ad50)

Reply:
Indeed, we should use `cli-kit` prompts to get the store password. However, we really need to rely on `readline` to power REPL DX, as `readline` also supports history of commands (when users press <kbd>ꜛ</kbd>), the minimal UI component cohesive with other REPL tools, and the life cycle callbacks.